### PR TITLE
address cron job confusion and update dependencies

### LINKED ISSUE #53: Schedule doesn't seem to work correctly
Hi, 

First of all, thanks for open sourcing this interesting project. 

I was playing with it and found out something odd. However, I am not completely sure if the issue is in your service, or in the cron library that you are using down the line.

I have set up a deployment using the following Schedule option:
```
- name: SCHEDULE
   value: "0 20 * * *"
```

My expectation was that `pod-reaper` would check for pods to 🔥  at 20h everyday. However, this is the result I am getting:

```
{"level":"info","msg":"loaded rule: chaos chance 0.999","time":"2020-03-26T16:04:16Z"}                                                                                    │
│ {"level":"info","msg":"reaping pod","pod":"service-2-764cdfbbdd-kft22","reasons":["was flagged for chaos"],"time":"2020-03-26T16:20:00Z"}     │
│ {"level":"info","msg":"reaping pod","pod":"service-1-5468d66b7b-wfqs2","reasons":["was flagged for chaos"],"time":"2020-03-26T16:20:00Z"}      │
│ {"level":"info","msg":"reaping pod","pod":"service-2-764cdfbbdd-mxcwg","reasons":["was flagged for chaos"],"time":"2020-03-26T17:20:00Z"}     │
│ {"level":"info","msg":"reaping pod","pod":"service-1-5468d66b7b-vx4kb","reasons":["was flagged for chaos"],"time":"2020-03-26T17:20:00Z"}      │
│ {"level":"info","msg":"reaping pod","pod":"service-2-5468d66b7b-cgq9c","reasons":["was flagged for chaos"],"time":"2020-03-26T18:20:00Z"}     │
│ {"level":"info","msg":"reaping pod","pod":"service-1-5468d66b7b-7zlv9","reasons":["was flagged for chaos"],"time":"2020-03-26T18:20:00Z"}      │
│ {"level":"info","msg":"reaping pod","pod":"service-2-764cdfbbdd-rkcmh","reasons":["was flagged for chaos"],"time":"2020-03-26T19:20:00Z"}     │
│ {"level":"info","msg":"reaping pod","pod":"service-1-5468d66b7b-cgq9c","reasons":["was flagged for chaos"],"time":"2020-03-26T19:20:00Z"}      │
│ {"level":"info","msg":"reaping pod","pod":"service-2-asfasfsasf-fsfds","reasons":["was flagged for chaos"],"time":"2020-03-26T20:20:00Z"}     │
│ {"level":"info","msg":"reaping pod","pod":"service-1-5468d66b7b-2mcpp","reasons":["was flagged for chaos"],"time":"2020-03-26T20:20:00Z"}
```

Is there anything that I am doing incorrectly? 

Thanks

### PR DESCRIPTION
**Pull request by brianberzins**
resolves #53 (thanks for bringing it to my attention @jnavarro86  

There are a lot of details to this is #53, but the summary is that the cron library being used defaulted to a parser that used seconds (required) with day of week being optional. In newer versions of the library, the default parser still uses seconds, but includes better documentation and usability for making custom parse formats. I changed the parsing format to include optional seconds with required day-of-week, which is closer to the standard (differing only by inclusion of optional seconds).

This change should help alleviate any confusion.
`* * * * *` now complies with standard cron (starting with minutes)
`* * * * * *` starts with seconds and includes day of week

Before this change the behavior is actually:
`* * * * *` starts with seconds and does not include a day of week
`* * * * * *` starts with seconds and includes day of week

This is arguably a breaking change, or arguably just a bug fix. I'm going with the later given my own expectations of how it worked.

Along the way, I need to update one dependency, so I just updated a number. Credit to @jdharmon for doing the heavy lifting on dependency updates -- this should be compatible with the changes to #50 

What's included:
- update changelog (this had fallen slightly behind)
- update golang version (which I ended up doing for local testing)
- update the readme to document the new optional seconds on cron schedule
- update vender dependencies (this includes the k8s dependencies that have been renamed)
- use a custom parser to get optional seconds (for better compatibility and more accurate cron schedules)

What could "break"?

The case where things changes is when someone was using `* * * * *` . For anyone doing this, this PR will change that behavior to matching the standard cron format. If someone was intentionally relying on that being non-standard, then this might cause some minor pain in either the reaper running less frequently, or no longer being a valid cron format (which should error). I think this is reasonable to be considered as a fix rather than a breaking change.

Paging @slushpupie for a review :smile:

### CODE REVIEW

**Comment by brianberzins** [comment]
Should mention that I got all paranoid about this and run a number of local tests (using example files) to make sure that everything was being run exactly when I expected it to.

File: glide.yaml [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
-  version: c1b2a29e2e9d121b9b7d818cb0d84d8098783236
+  version: =v0.17.0
   subpackages:
-    - /pkg/labels
-    - /pkg/selection
+  - pkg/labels
+  - pkg/selection
 - package: k8s.io/client-go
-  version: =v2.0.0
+  version: =v0.17.0
```
Comment (slushpupie) [inline glide.yaml]: Am I reading this right that we are going _backwards_ in version number here?

Comment (brianberzins) [reply glide.yaml]: I know it works, but that's a REALLY good question.

Comment (slushpupie) [reply glide.yaml]: Looking at the `client-go` versions, it looks like they switched mechanisms to match k8s versions, so `0.17.0` maps to `1.17.0`.  So it does appear this is a newer version, though not obviously so.

Comment (brianberzins) [reply glide.yaml]: I didn't have a chance to look before you, but I was *hoping* it'd be something like that... Doing to cause a mess if k8s ever get to 2.0.0!

**Comment by brianberzins** [comment]
Thank you @slushpupie !

**Pull request merged**